virtualizarr/manifests/store.py

```diff
@@ -98,14 +98,41 @@ def parse_manifest_index(
     return tuple(int(ind) for ind in chunk_component.split(chunk_key_encoding))
 
 
-def get_deepest_group_or_array(
-    node: ManifestGroup, key: Iterable[str]
-) -> ManifestGroup | ManifestArray:
-    for var in key:
+def _get_deepest_group_or_array(
+    node: ManifestGroup | ManifestArray, key: str
```

Comment:
It looks like `node` will always be a `ManifestGroup`, no?

```suggestion
    node: ManifestGroup, key: str
```